cilium-cli/connectivity/builder/strict_mode_encryption.go

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package builder
+
+import (
+	"github.com/cilium/cilium/cilium-cli/connectivity/check"
+	"github.com/cilium/cilium/cilium-cli/connectivity/tests"
+	"github.com/cilium/cilium/cilium-cli/utils/features"
+)
+
+type strictModeEncryption struct{}
+
+func (t strictModeEncryption) build(ct *check.ConnectivityTest, _ map[string]string) {
+	println("strictModeEncryption")
+	newTest("strict-mode-encryption", ct).
+		WithCondition(func() bool { return ct.Params().IncludeUnsafeTests }).
+		WithFeatureRequirements(
+			features.RequireEnabled(features.EncryptionStrictMode),
+			// Strict mode is only supported with WireGuard
+			features.RequireMode(features.EncryptionPod, "wireguard"),
+			// Strict mode always allows host-to-host tunnel traffic
+			features.RequireDisabled(features.Tunnel),
+		).
+		WithScenarios(tests.PodToPodMissingIPCache()).
+		WithExpectations(func(_ *check.Action) (egress, ingress check.Result) {
+			return check.ResultDrop, check.ResultDrop
```

Comment:
Would it make sense to define a custom result that contains the drop reason (see e.g. `check.ResultDefaultDenyEgressDrop`)?